llpc/util/llpcElfWriter.cpp

```diff
@@ -997,7 +1003,7 @@ void ElfWriter<Elf>::Reinitialize()
 // =====================================================================================================================
 // Link the relocatable ELF readers into a pipeline ELF.
 template<class Elf>
-Result ElfWriter<Elf>::LinkRelocatableElf(
+Result ElfWriter<Elf>::LinkGraphicsRelocatableElf(
```

Comment:
If you plan to support vs+fs only, please don't use arrayref in function parameters, instead, please use vs, fs as separate function parameters.
if you plan to support all graphics shader stage in the future, I prefer keep each shader stage with fixed index.
i.e. vs = relocatableElfs[0], fs=relocatableElfs[4], and etc.

Reply:
I disagree with this. Since Microsoft added new kinds of shader stages to DX12, those are likely to come to Vulkan in the future as well, and hard-coding shader stage indices makes adding support for these other shader stages more prone to code integration problems.

I think it makes sense to allow shader stages to be provided in the order of the pipeline. Importantly, we do need to document what it is that functions expect.

Reply:
@nhaehnle Please advise if there is any actionable items.

Reply:
The problem is we can't get shader stage info from ElfReader<Elf> now. so if we use an array of shaders in the order of the pipeline, we don't know which stages are valid in the pipeline.

Reply:
We could get the shader stage from the elf by looking at the pal metadata.

Reply:
Do we have a conclusion on this?

Reply:
The hunk is good as-is, but give Qun a chance to respond. One of the features of Git is that you *can* have multiple changes in flight easily, so while it's good to keep moving forward, we don't *have* to rush things :)

Generally speaking, I insist that the magic sparse array is not a good way of doing things. If a compacted array doesn't work out, we can use a sparse struct instead, for example (the struct would have vs, tcs, tes, etc. members). It's premature to worry about that here though.

Reply:
I am OK if you plan to check the PAL metadata  to get stage info in the future.